src/lightning_app/core/queues.py

```diff
@@ -377,6 +408,13 @@ def length(self):
         val = self.client.get(f"/v1/{self.app_id}/{self._name_suffix}/length")
         return int(val.text)
 
+    def get_all(self, timeout: float):
+        time.sleep(timeout)
+        resp = self.client.post(f"v1/{self.app_id}/{self._name_suffix}", query_params={"action": "lrange"})
```

Comment:
`lrange` sort of leaks the redis implementation detail. I would just add `popAll`

Reply:
or `batchPop` with another argument `batchSize`, so we can limit the number of elements returned

Reply:
This part is for you to implement. You can put the API you want ;)